Add pca post (#41)
haam-community/haam-community.github.io: _posts/2024-04-14-blog.md

## Introduction

In the field of ancient human population genetics, we often analyse ancient genomes with high proportions of missing data, namely in the recovered genotypes.
This can stem from limitations of material preservation, and/or of available laboratory/sequencing technologies.
Given that sampling remains for ancient DNA is a destructive process, we have an ethical responsibility to make the most of the data we are able to generate.
At the same time, we need to know exactly the limitations of our data when inferring results, and present our findings fairly and compassionately.
After all, the past breeds the present, and the interpretations we make of the past can have implications affecting present-day peoples and communities.

To that end, archaeogenticists often use quality cutoff thresholds in our analyses, limiting the set of analysed individuals to ones with sufficient SNP coverage for population genetics analyses.
These cutoffs are often expressed in terms of the "Number of covered SNPs", i.e. the number of SNPs which do not have a missing genotype call for a given ancient individual.
But what constitutes a sensible minimum cutoff?

> Principal Component Analysis (PCA) is a very useful _visualisation_ tool for human aDNA data, as well as the source of many-a [popgen bingo](https://ancientpopgen-bingo.firebaseapp.com/?game=ancientpopgen-bingo).

Reviewer: Maybe show classic Jhohn novembre plot or something to get people in the right mind?

> It is often the case that most (if not all) sampled individuals will be featured in a PCA plot, and it is important to understand how missing data affects their position in order to avoid misinterpreting PCA results.

Reviewer: ```suggestion
It is often the case that most (if not all) individuals in an ancient population genetics study will be featured in a PCA plot, and it is important to understand how missing data affects their position in order to avoid misinterpreting PCA results.
```

So, I set out to find out:

How does missing data affect the projected position of individuals on a PCA plot?
Can you trust the placement of that low coverage individual in the paper you're reviewing, and if so to what degree?

Let's find out!

## Experimental setup

I started with a dataset of present day individuals on the Human Origins array (\~600k SNPs) <sup>[[1](http://dx.doi.org/10.1534/genetics.112.145037),[2](http://dx.doi.org/10.1038/nature13673),[3](https://doi.org/10.1038/s41559-019-0878-2)]</sup>.
I picked a set of candidate present-day individuals (plus the 'Loschbour' hunter-gatherer) whose positions roughly spanned the area of each PCA plot.
I then randomly replaced the existing genotypes with missing ones for each candidate individual at a given rate, creating 200 replicates per missingness rate, per individual.

> I used smartpca <sup>[[4](http://dx.doi.org/10.1371/journal.pgen.0020190)]</sup> to run PCA and projected all replicates (lsqproject: YES) across all missingness rates for all candidate individuals.

Reviewer: ```suggestion
I used smartpca <sup>[[4](http://dx.doi.org/10.1371/journal.pgen.0020190)]</sup> to run PCA and projected all replicates (`lsqproject: YES`) across all missingness rates for all candidate individuals.
```

Reviewer: Don't really understand this specific technical detail when no other technical info

> The original candidate individuals were not projected, with the exception of 'Loschbour', which is also the only individual with pseudohaploid genotypes in the experiment.

Reviewer: ```suggestion
The original candidate individuals were not projected, with the exception of 'Loschbour'. Compared to the other individuals, it is also the only individual with data to have been generated with 'pseudohaploid' genotypes in the experiment.
```

I think this phrasing would make  it a bit clearer the relevance here. I would also give a link to what 'pseudohaploid' means.

> The 'Loschbour' hunter-gatherer can then act as a test case for true ancient data, where differences between the replicate spread of this individual to other candidate individuals could indicate that damage, genotyping procedure, and/or projecting individuals whose genetic variation is not directly included in the PC calculation can also affect the spread of data due to missingness.

Reviewer: ```suggestion
The 'Loschbour' hunter-gatherer can then act as a test case for true ancient data, where differences between the replicate spread of this individual to other candidate individuals could indicate that any or all of damage, genotyping procedure, and/or projecting individuals whose genetic variation is not directly included in the PC calculation can also affect the spread of data due to missingness.
```

I focused on four PCA setups that are somewhat prevalent in human aDNA literature:

-   Worldwide PCA

-   African PCA (à la [Skoglund et. al 2017](http://dx.doi.org/10.1016/j.cell.2017.08.049))

-   Eurasian PCA (à la [Jeong2019](https://doi.org/10.1038/s41559-019-0878-2))

-   West Eurasian PCA (à la [Lazaridis et al. 2014](http://dx.doi.org/10.1038/nature13673))

## Results

> In each slideshow below, you can see the results for eahc of the regional PCAs for missingness rates of 90%, 97%, and 99%, corresponding to approximately 60.000, 18.000, and 6.000 covered SNPs resprectively.

Reviewer: ```suggestion
In each slideshow below, you can see the results for each of the regional PCAs for missingness rates of 90%, 97%, and 99%, corresponding to approximately 60.000, 18.000, and 6.000 covered SNPs resprectively.
```

Present-day populations are shown in the background, plotted as a unique 3-4 letter code per population and coloured according to their approximate geographical location/region.
The "true" position of each candidate individual (i.e. their position when using all the available genotypes) is shown by black symbols.

> Around each of these symbols, the convex hull of the 200 replicates with artificailly-introduced missing genotypes is plotted, representing the variability of the projected placement of the replicates individuals.

Reviewer: ```suggestion
Around each of these symbols, the convex hull of the 200 replicates with artificially-introduced missing genotypes is plotted, representing the variability of the projected placement of the replicates individuals.
```

  <div class="mySlides slide2 fade">
    <div class="numbertext">1 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.9.png" | relative_url }}" style="width:90%">
    <br>
    <div class="text">Spread of replicates with 90% of genotypes replaced by missing data (~ 60.000 SNPs left).</div>
  </div>

  <div class="mySlides slide2 fade">
    <div class="numbertext">2 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.97.png" | relative_url }}" style="width:90%">
   <br>
    <div class="text">Spread of replicates with 97% of genotypes replaced by missing data (~ 18.000 SNPs left).</div>
  </div>

  <div class="mySlides slide2 fade">
    <div class="numbertext">3 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.99.png" | relative_url }}" style="width:90%">
   <br>
    <div class="text">Spread of replicates with 99% of genotypes replaced by missing data (~ 6.000 SNPs left).</div>
  </div>

  <!-- Next and previous buttons -->
  <a class="prev" onclick="plusSlides(-1, 2)">&#10094;</a>
  <a class="next" onclick="plusSlides(1, 2)">&#10095;</a>
</div>
<br>

## Conclusions

> In literature, the chosen cutoffs are both sensible and (strictly speaking) arbitrary, commonly ranging between 10.000 and 20.000 covered SNPs.

Reviewer: ```suggestion
In the ancient population-genetics literature, the commonly chosen cutoffs are both sensible and (strictly speaking) arbitrary, typically ranging between 10.000 and 20.000 covered SNPs.
```

Any individuals with fewer than that number of SNPs are often excluded from downstream analyses.
Looking at the results above, you can see why:

>  - The degree of variation in placement of the replicates at high missingness rates can make inferences of local ancestry based on PCA erroneous.

Reviewer: ```suggestion
 - The degree of variation in placement of the replicates at high missingness rates can make inferences of _local_ ancestry based on PCA erroneous.
```

These cutoffs are not absolute, however. There are still things we can infer about individuals with a low number of covered SNPs:

 - While the exact projection of low-coverage individuals in the Worldwide PCA is also affected, the spread of replicates shows that it is still perfectly possible to assess if a studied individual is genetically more closely related to present-day African, South Asian or Native American populations from the PCA results, even from only a few thousand SNPs.
   - While not shown here, it should be noted that differentiating between Asian and Native American populations (whose positions are very similar in the Worldwide PCA plot above), will be possible when considering additional PCs. 

An additional thing to note about the results from this analysis (especially when looking at the West Eurasian PCA) is that the spread of replicates around the 'Loschbour' hunter-gatherer is not visibly different for that around other candidate individuals. 

> This suggests that damage artefacts, the pseudohaploid genotyping process, as well as the fact that this individual is not in the 'poplist' of the PCA (and hence its variation is not used in the calculation of the PCs), do not seem to exacerbate the spread of replicates.

Reviewer: I think referring to it was projected (unlike the others, if I understand) would make more sense. You mention projection in the exeception above but not here, and then introduce the 'poplist' here which is otherwise not mentioned

After all, the past breeds the present, and the interpretations we make of the past can have implications affecting present-day peoples and communities.

To that end, archaeogenticists often use quality cutoff thresholds in our analyses, limiting the set of analysed individuals to ones with sufficient coverage for population genetics analyses.
These cutoffs are often expressed in terms of the "Number of covered SNPs", i.e. the number of SNPs which do not have a missing genotype call for a given ancient individual.
But what constitutes a sensible minimum cutoff?

It is often the case that most (if not all) sampled individuals will be featured in a Principal Component Analysis (PCA), which is both a very useful visualisation tool for human aDNA data, and the source of many-a [popgen bingo](https://ancientpopgen-bingo.firebaseapp.com/?game=ancientpopgen-bingo).
How does missing data affect the projected position of ancient individuals on a PCA plot?
Can you trust the placement of that low coverage individual in the paper you're reviewing, and if so to what degree?

Let's find out!

## Experimental setup

> I started with a dataset of present day individuals on the Human Origins array (\~600k SNPs) <sup>[[1](http://dx.doi.org/10.1534/genetics.112.145037),[2](http://dx.doi.org/10.1038/nature13673),[3](http://dx.doi.org/10.1534/genetics.112.145037)]</sup>.

Author: ```suggestion
I started with a dataset of present day individuals on the Human Origins array (\~600k SNPs) <sup>[[1](http://dx.doi.org/10.1534/genetics.112.145037),[2](http://dx.doi.org/10.1038/nature13673),[3](https://doi.org/10.1038/s41559-019-0878-2)]</sup>.
```

I picked a set of candidate present-day individuals (plus the 'Loschbour' hunter-gatherer) whose positions roughly spanned the area of each PCA plot.
I then randomly added missing data to each candidate individual's genotypes at a given rate, creating 200 replicates per missingness rate, per individual.

> 

Author: ```suggestion
I then used smartpca [4](http://dx.doi.org/10.1371/journal.pgen.0020190) to run PCA and projected all replicates (`lsqproject: YES`) across all missingness rates for all candidate individuals. The original candidate individuals were not projected, with the exception of 'Loschbour', which is also the only individual with pseudohaploid genotypes in the experiment.

```

I focused on four PCA setups that are somewhat prevalent in human aDNA literature:

-   Worldwide PCA

> -   Eurasian PCA (à la [Jeong2019](http://dx.doi.org/10.1534/genetics.112.145037))

Author: ```suggestion
-   Eurasian PCA (à la [Jeong2019](https://doi.org/10.1038/s41559-019-0878-2))
```

### Missing 97% of all genotypes (18.000 SNPs)

| :-- | --: |
![Worldwide](/assets/media/event_images/2024-04-14-blog/Worldwide.m0.97.png) |![Eurasian](/assets/media/event_images/2024-04-14-blog/Eurasia.m0.97.png)
![West Eurasian](/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.97.png) |![African](/assets/media/event_images/2024-04-14-blog/Africa.m0.97.png)

### Missing 99% of all genotypes (6.000 SNPs)

| :-- | --: |
![Worldwide](/assets/media/event_images/2024-04-14-blog/Worldwide.m0.99.png) |![West Eurasian](/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.99.png)

## Conclusions

In literature, the chosen cutoffs are both sensible and (strictly speaking) arbitrary, commonly ranging between 10.000 and 20.000 covered SNPs.
Any individuals with fewer than that number of SNPs are often excluded from downstream analyses.
Looking at the results above, you can see why:

 - High degrees of missingness can affect the projected placement of an individual in regional PCAs. 
    - e.g. with sparse enough data (99% missing), some replicates of the Sicilian individual 'TP08' are projected closer to present-day Greeks than the other Sicilians in the dataset.

>  - While the exact projection of low-coverage individuals in the Worldwide PCA is also affected, the spread of replicates shows that it is still perfectly possible to access if a studied individual is genetically more closely related to present-day African, South Asian or Native American populations from the PCA results, even from only a few thousand SNPs.

Author: ```suggestion
 - While the exact projection of low-coverage individuals in the Worldwide PCA is also affected, the spread of replicates shows that it is still perfectly possible to assess if a studied individual is genetically more closely related to present-day African, South Asian or Native American populations from the PCA results, even from only a few thousand SNPs.
```

>  - Comparing the spread of replicates around 'Loschbour' to those around the other candidate individuals, it does not look like projection of the candidate individual itself affects the spread of the replicates. (n=1!!)

Author: Clarify/rewrite this

Reviewer: Agree, I would put point 1 in it's own 'list'.

You sort of have two categories of outcomes:

1. For in-depth technical analyses, there is quite a lot of variatoin in placement (which is why there are cut offs and these are important)

But at the same time

1. The variation isn't enough to give you a _rough_ idea of where it falls in broader conexts
2. The candidate project doesn't effect spread of replicates (I don't get what you mean by this one, but I'm not a poop-genner)

---
layout: post
title: "Missing data and PCA"
subtitle: "How much is too much?"
authors:
 - Thiseas C. Lamnidis
categories: Blog
tag: Blog
---

_by [Thiseas C. Lamnidis](https://www.eva.mpg.de/archaeogenetics/staff/thiseas-christos-lamnidis/)_

## Introduction

In the field of ancient human population genetics, we often analyse ancient genomes with high proportions of missing genotypes.
This can stem from limitations of material preservation, and/or of available laboratory/sequencing technologies.
Given that sampling remains for ancient DNA is a destructive process, we have the ethical responsibility to make the most of the data we are able to generate.
Additionally, we need to know exactly the limitations of our data when inferring results, and present our findings fairly and compassionately.

> After all, the past breeds the present, and the interpretations we make of the past can have implications affecting present-day peoples and communities.

Author: I quite like the original sentence/paragraph tbh.

To that end, archaeogenticists often use quality cutoff thresholds in our analyses, limiting the set of analysed individuals to ones with sufficient coverage for population genetics analyses.
These cutoffs are often expressed in terms of the "Number of covered SNPs", i.e. the number of SNPs which do not have a missing genotype call for a given ancient individual.

> But what constitutes a sensible minimum cutoff?

Author: Not that I can recall (at least within popgen. Likely there will be linear algebra papers about it somewhere.). hence the post

It is often the case that most (if not all) sampled individuals will be featured in a Principal Component Analysis (PCA), which is both a very useful visualisation tool for human aDNA data, and the source of many-a [popgen bingo](https://ancientpopgen-bingo.firebaseapp.com/?game=ancientpopgen-bingo).
How does missing data affect the projected position of ancient individuals on a PCA plot?
Can you trust the placement of that low coverage individual in the paper you're reviewing, and if so to what degree?

Let's find out!

## Experimental setup

I started with a dataset of present day individuals on the Human Origins array (\~600k SNPs) <sup>[[1](http://dx.doi.org/10.1534/genetics.112.145037),[2](http://dx.doi.org/10.1038/nature13673),[3](https://doi.org/10.1038/s41559-019-0878-2)]</sup>.
I picked a set of candidate present-day individuals (plus the 'Loschbour' hunter-gatherer) whose positions roughly spanned the area of each PCA plot.

> I then randomly added missing data to each candidate individual's genotypes at a given rate, creating 200 replicates per missingness rate, per individual.

Reviewer: Added, or replaced existing genotypes with missing?

> I then used smartpca [4](http://dx.doi.org/10.1371/journal.pgen.0020190) to run PCA and projected all replicates (`lsqproject: YES`) across all missingness rates for all candidate individuals. The original candidate individuals were not projected, with the exception of 'Loschbour', which is also the only individual with pseudohaploid genotypes in the experiment.

Reviewer: Maybe wrap 
```suggestion
I then used smartpca ([4](http://dx.doi.org/10.1371/journal.pgen.0020190)) to run PCA and projected all replicates (`lsqproject: YES`) across all missingness rates for all candidate individuals. The original candidate individuals were not projected, with the exception of 'Loschbour', which is also the only individual with pseudohaploid genotypes in the experiment.
```

Maybe wrap citations in quotes? Renders a bit wierd otherwise...

Reviewer: And no actual commands or code?

Reviewer: What does pseudohaploid genotypes got to do with anything?

> I focused on four PCA setups that are somewhat prevalent in human aDNA literature:

Reviewer: I would have mentioned this earlier

-   Worldwide PCA

> -   Eurasian PCA (à la [Jeong2019](https://doi.org/10.1038/s41559-019-0878-2))

Reviewer: ```suggestion
-   Eurasian PCA (e.g. [Jeong2019](https://doi.org/10.1038/s41559-019-0878-2))
```

-   West Eurasian PCA (à la [Lazaridis et al. 2014](http://dx.doi.org/10.1038/nature13673))

-   African PCA (à la [Skoglund et. al 2017](http://dx.doi.org/10.1016/j.cell.2017.08.049))

> ## Results

Reviewer: I would spend some time here describing what the PCAs are made up of. You suddenly mention in the very last paragraph something about hull,s but you never mention that is how the replicates are represented

(or you write figure captions, which serve the same purpose).

> <sup>(You can right click on any of the images below and select "Open Image in a New Tab" to see them in more detail.)</sup>

Reviewer: Is this HTML code for superscript? Could you use around citatoins if so

Author: it is, and I do

### Missing 90% of all genotypes (60.000 SNPs)

| :-- | --: |
![Worldwide](/assets/media/event_images/2024-04-14-blog/Worldwide.m0.9.png)| ![Eurasian](/assets/media/event_images/2024-04-14-blog/Eurasia.m0.9.png)
![West Eurasian](/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.9.png)|![African](/assets/media/event_images/2024-04-14-blog/Africa.m0.9.png)

### Missing 97% of all genotypes (18.000 SNPs)

| :-- | --: |
![Worldwide](/assets/media/event_images/2024-04-14-blog/Worldwide.m0.97.png) |![Eurasian](/assets/media/event_images/2024-04-14-blog/Eurasia.m0.97.png)
![West Eurasian](/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.97.png) |![African](/assets/media/event_images/2024-04-14-blog/Africa.m0.97.png)

### Missing 99% of all genotypes (6.000 SNPs)

| :-- | --: |

> ![Worldwide](/assets/media/event_images/2024-04-14-blog/Worldwide.m0.99.png) |![West Eurasian](/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.99.png)

Reviewer: Generally I find the changing axis scale lengths very jarring and difficult to read.

Also if I'm comparing  the effect of increasing missingness, I want the same PCA-type next to each  other (so worldwide @ 90, 97, 99) all together.

It's currently very difficult to actually compared. It would also make the axis length changes less jarring.

Also missing proper alt_text for each plot (important for accessibility purposes - something I'm being picky with as I've only myself taking more time to do this recently :sweat_smile: _
